manifests/init.pp

```diff
@@ -39,4 +41,6 @@
     group  => 'systemd-network',
     *      => $options,
   }
+
+  create_resources(wireguard::interface, $interfaces)
```

Comment:
instead if using create_resources, can you iterate 9n the hash with .each and instantiate wireguard::interface for each hash entry?